java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture6/tasks/_3/UserService.java

```diff
@@ -0,0 +1,11 @@
+package com.zoolatech.lecture6.tasks._3;
+
+record UserService(UserRepository userRepository) {
+
+    public String findUserEmail(String id) throws UserMissingException {
+        if (!userRepository.findUserEmail(id).isEmpty()) {
+            return userRepository.findUserEmail(id);
+        }
+        return "No such user in cash and table";
```

Comment:
A caller of this method expects to receive an email. `No such user in cash and table` doesn't look like an email and can be unexpected result. It's better to return null.